atc/db/resource_type.go

```diff
@@ -48,16 +48,15 @@ type ResourceType interface {
 
 	SetResourceConfigScope(ResourceConfigScope) error
 
-	CheckPlan(atc.Version, time.Duration, ResourceTypes, atc.Source) atc.CheckPlan
+	CheckPlan(planFactory atc.PlanFactory, imagePlanner ImagePlanner, from atc.Version, interval time.Duration, sourceDefaults atc.Source, manuallyTriggered bool, shallow bool) atc.Plan
 	CreateBuild(context.Context, bool, atc.Plan) (Build, bool, error)
 
-	Version() atc.Version
-
 	Reload() (bool, error)
 }
 
 type ResourceTypes []ResourceType
 
+// XXX: TODO Can remove the bool
```

Comment:
Meh, not sure if it's necessary. Returning the `bool` at least forces callers to deal with the fact that there may be no parent - it's possible to forget to check `!= nil`